firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -199,7 +196,40 @@ void runTransaction(String action, Runnable operation) {
   }
 
   long getByteSize() {
-    return databasePath.length();
+    return getPageCount() * getPageSize();
+  }
+
+  /**
+   * Gets the page size of the database. Typically 4096.
+   *
+   * @see https://www.sqlite.org/pragma.html#pragma_page_size
+   */
+  private long getPageSize() {
+    return pragma("page_size");
+  }
+
+  /**
+   * Gets the number of pages in the database file. Multiplying this with the page size yields the
+   * approximate size of the database on disk (including the WAL, if relevant).
+   *
+   * @see https://www.sqlite.org/pragma.html#pragma_page_count.
+   */
+  private long getPageCount() {
+    return pragma("page_count");
+  }
+
+  /**
+   * @param pragmaName Must be a valid 'pragma-name' and must refer to a long value.
+   * @see https://www.sqlite.org/pragma.html
+   */
+  private long pragma(String pragmaName) {
+    Cursor cursor = db.rawQuery("PRAGMA " + pragmaName, null);
```

Comment:
As a matter of principle we should avoid composing SQL through string concatenation. This construct is subject to SQL injection attacks if we're not careful. Better to just have the caller pass the whole query string.

Reply:
As discussed, it's not possible to use a prepared statement here. We could instead allow the caller to pass the *entire* sql statement, but that's probably worse. Instead, we decided to un-refactor this. (Done).